main/utils/textEntry.c

```diff
@@ -71,215 +44,74 @@ static const char keyboard_lower[] = "\
 
 static const uint8_t lengthperline[] = {14, 14, 13, 12, 1};
 
-/*============================================================================
- * Functions
- *==========================================================================*/
+//==============================================================================
+// Functions
+//==============================================================================
 
-/**
- * Initialize the text entry
- *
- * @param usefont The font to use, should be ibm_vga8
- * @param max_len The length of buffer
- * @param buffer  A char* to store the entered text in
- */
 void textEntryStart(font_t* usefont, int max_len, char* buffer)
 {
-    texLen       = max_len;
-    texString    = buffer;
-    selx         = 1;
-    sely         = 1;
-    keyMod       = NO_SHIFT;
-    texString[0] = 0;
-    cursorTimer  = 0;
-    textEntryIBM = usefont;
+    texLen         = max_len;
+    texString      = buffer;
+    selx           = 1;
+    sely           = 1;
+    keyMod         = NO_SHIFT;
+    texString[0]   = 0;
+    cursorTimer    = 0;
+    cursorToggle   = true;
+    activeFont     = usefont;
+    textColor      = WHITE;
+    empahsisColor  = WHITE;
+    prettyGraphics = false;
 }
 
-/**
- * Finish the text entry by disarming the cursor blink timer
- */
-void textEntryEnd(void)
+void textEntryStartPretty(font_t* usefont, int max_len, char* buffer, wsg_t* BG)
 {
+    texLen         = max_len;
+    texString      = buffer;
+    selx           = 1;
+    sely           = 1;
+    keyMod         = NO_SHIFT;
+    texString[0]   = 0;
+    cursorTimer    = 0;
+    cursorToggle   = true;
+    activeFont     = usefont;
+    prettyGraphics = true;
+    bgImage        = BG;
 }
 
-/**
- * Draw the text entry UI
- *
- * @return true if text entry is still being used
- *         false if text entry is finished
- */
-bool textEntryDraw(void)
+bool textEntryDraw()
+{
+    return textEntryDrawBlink(0);
+}
+
+bool textEntryDrawBlink(int64_t elapsedUs)
 {
     // If we're done, return false
     if (keyMod == SPECIAL_DONE)
     {
         return false;
     }
-
-    // Draw the text entered so far
+    if (prettyGraphics)
     {
-        const int16_t text_h = 32;
-        int16_t textLen      = textWidth(textEntryIBM, texString) + textEntryIBM->chars[0].width;
-        int16_t endPos       = drawText(textEntryIBM, WHITE, texString, (TFT_WIDTH - textLen) / 2, text_h);
-
-        // If the blinky cursor should be shown, draw it
-        if ((cursorTimer++) & 0x10)
-        {
-            drawLineFast(endPos + 1, text_h - 2, endPos + 1, text_h + textEntryIBM->height + 1, WHITE);
-        }
+        // Background
```

Comment:
So, a custom, default wsg, a blank color, and using the maniaRenderer. Understood.

Reply:
I don't think you need a default WSG or to use `menuManiaRenderer` explicitly. In fact, you don't want a separate `menuManiaRenderer` from what the mode has because the animations won't sync up. For example, check out Ultimate TTT's "Marker Select" to see some nice overdraw with consistent animation. The list of background options should be
1. Custom WSG
1. Transparent (i.e. don't draw a background)
1. Solid color
    * Default solid color is `c000`